sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -377,6 +378,39 @@ case class InsertIntoTable(
   override lazy val resolved: Boolean = childrenResolved && table.resolved
 }
 
+/** Factory for constructing new `View` nodes. */
+object View {
+  def apply(desc: CatalogTable): View = View(desc, desc.schema.toAttributes, None)
+}
+
+/**
+ * A container for holding the view description(CatalogTable), and the output of the view. The
+ * child will be defined if the view is resolved with Hive support, else it should be None.
+ * This operator will be removed at the end of analysis stage.
+ *
+ * @param desc A view description(CatalogTable) that provides necessary information to resolve the
+ *             view.
+ * @param output The output of a view operator, this is generated during planning the view, so that
+ *               we are able to decouple the output from the underlying structure.
+ * @param child The logical plan of a view operator, it should be non-empty if the view is resolved
+ *              with Hive support, else it should be None.
+ */
+case class View(
+    desc: CatalogTable,
+    output: Seq[Attribute],
```

Comment:
Why can't we just use `def output = child.output`? If we wanna reorder the columns according to the original view schema, we can wrap the child with `Project`, e.g.
```
// The relation is a view, so we wrap the relation by:
// 1. Add a [[View]] operator over the relation to keep track of the view desc;
// 2. Wrap the logical plan in a [[SubqueryAlias]] which tracks the name of the view.
val viewPlan = sparkSession.sessionState.sqlParser.parsePlan(viewText)
val child = View(desc = table, child = Some(Project(schema.map(f => UnresolveAttribute(Seq(f.name)))), viewPlan))
SubqueryAlias(alias.getOrElse(table.identifier.table), child, Option(table.identifier))
```